src/post.js

```diff
@@ -10,12 +10,42 @@ module.exports = async config => {
   const name = `v${pkg.version}`;
   const tag = {owner, repo, ref: `refs/tags/${name}`, sha: await promisify(gitHead)()};
   const body = await promisify(plugins.generateNotes)(config);
-  const release = {owner, repo, tag_name: name, name, target_commitish: branch, draft: !!debug, body};
+  let githubPreRelease = false;
+  let release = {
+    owner,
+    repo,
+    tag_name: name,
+    name,
+    target_commitish: branch,
+    draft: !!debug,
+    body,
+    prerelease: githubPreRelease,
+  };
 
   if (debug && !githubToken) {
     return {published: false, release};
   }
 
+  // obtain npm dist tag if any
```

Comment:
The tests in `post.test.js` are fairly complex (use `nock`, call a bunch of submodules etc), so it would probably better to externalize the determination of the `prerelease` value to a sub-module in `src/lib`. You have seral example there.
This way you could create a test file dedicated to this sub-module that would be simpler and more focused. This way you would test there all the details regarding the determination of `prerelease` (default parameters, parameter with a value, dist-tag with value x, y, z etc) and in `post.test.js` you would just need one test that will only validate that whatever value returned by the sub-module is properly sent to gihub and that `post.js` pass the correct parameter to the sub-module.

Reply:
I'm trying that at the moment. Is there any good way of debugging these tests? Or at least a way of printing something to the console.

Reply:
`console.log` would print to the console when you run the tests.
You can run `ava -v --watch` so each test of source file you modify will immediately run so that make it easier.
You can also temporarily add `only` to your test (for example `test.only('test title' ... )` or `test.serial.only('test title' ... )`), so only this test will run.
You can find more info on Ava here https://github.com/avajs/ava